src/tests/ftest/cart/test_group_np_common.h

```diff
@@ -146,48 +149,43 @@ test_swim_status_handler(crt_rpc_t *rpc_req)
 	regex_t				 regex_dead;
 	static const char		*dead_regex = ".?0*1";
 	static const char		*alive_regex = ".?0*";
-	char				swim_seq[MAX_SWIM_STATUSES];
-	size_t				sc = sizeof(char);
+	int				rank;
 
 	/* CaRT internally already allocated the input/output buffer */
 	e_req = crt_req_get(rpc_req);
-
-	if (swim_status_by_rank[e_req->rank] != NULL)
-		memcpy(swim_seq,
-		       swim_status_by_rank[e_req->rank],
-		       strlen(swim_status_by_rank[e_req->rank]) * sc);
-	else
-		memset(swim_seq, 0x0, MAX_SWIM_STATUSES);
+	rank = e_req->rank;
 
 	/* compile and run regex's */
 	regcomp(&regex_dead, dead_regex, REG_EXTENDED);
```

Comment:
Memory leak, must free regex_dead.

Reply:
Where is `regex_dead` `malloc`'d?  For this small example test-regex.c program:
```c
#include <regex.h>
#include <stddef.h>
#include <stdlib.h>

#define A 1024
#define B 1024

static char s[A][B] = { { 0 } };

int main() {
        regex_t r;
        static const char *ar = ".?0*";
        regcomp(&r, ar, REG_EXTENDED);
        int rc_alive = regexec(&r, s[3], 0, NULL, 0);
        free(&r);

}
```
-- gcc reports double-free:

```
$ gcc test-regex.c && ./a.out
test-regex.c: In function ‘main’:
test-regex.c:15:6: warning: attempt to free a non-heap object ‘r’ [-Wfree-nonheap-object]
  free(&r);
      ^
*** Error in `./a.out': double free or corruption (out): 0x00007fff736fe260 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x81299)[0x7f8473575299]
./a.out[0x400630]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7f8473516555]
./a.out[0x400519]
======= Memory map: ========
00400000-00401000 r-xp 00000000 00:2c 46427745                           a.out
00600000-00601000 r--p 00000000 00:2c 46427745                           a.out
00601000-00602000 rw-p 00001000 00:2c 46427745                           a.out
00602000-00702000 rw-p 00000000 00:00 0
0103f000-01060000 rw-p 00000000 00:00 0                                  [heap]
7f846c000000-7f846c021000 rw-p 00000000 00:00 0
7f846c021000-7f8470000000 ---p 00000000 00:00 0
7f84732de000-7f84732f3000 r-xp 00000000 fd:01 1048595                    /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7f84732f3000-7f84734f2000 ---p 00015000 fd:01 1048595                    /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7f84734f2000-7f84734f3000 r--p 00014000 fd:01 1048595                    /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7f84734f3000-7f84734f4000 rw-p 00015000 fd:01 1048595                    /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7f84734f4000-7f84736b7000 r-xp 00000000 fd:01 1049590                    /usr/lib64/libc-2.17.so
7f84736b7000-7f84738b7000 ---p 001c3000 fd:01 1049590                    /usr/lib64/libc-2.17.so
7f84738b7000-7f84738bb000 r--p 001c3000 fd:01 1049590                    /usr/lib64/libc-2.17.so
7f84738bb000-7f84738bd000 rw-p 001c7000 fd:01 1049590                    /usr/lib64/libc-2.17.so
7f84738bd000-7f84738c2000 rw-p 00000000 00:00 0
7f84738c2000-7f84738e4000 r-xp 00000000 fd:01 1049583                    /usr/lib64/ld-2.17.so
7f8473abd000-7f8473ac0000 rw-p 00000000 00:00 0
7f8473ae1000-7f8473ae3000 rw-p 00000000 00:00 0
7f8473ae3000-7f8473ae4000 r--p 00021000 fd:01 1049583                    /usr/lib64/ld-2.17.so
7f8473ae4000-7f8473ae5000 rw-p 00022000 fd:01 1049583                    /usr/lib64/ld-2.17.so
7f8473ae5000-7f8473ae6000 rw-p 00000000 00:00 0
7fff736df000-7fff73701000 rw-p 00000000 00:00 0                          [stack]
7fff73792000-7fff73794000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Aborted
```

Reply:
regex_t is defined as a pointer.  When you call regcomp, you are passing a pointer to a pointer.  That functions allocates memory for a structure and returns it to you.  At the end, you need to call regfree for that pointer.  There are a few places in the test code that does that.

Reply:
Ah, I see - thank you for clarifying.  cd7d209